files/image_config/topology/topology.sh

```diff
@@ -7,15 +7,33 @@
 
 start() {
     TOPOLOGY_SCRIPT="topology.sh"
-    PLATFORM=${PLATFORM:-`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`}
-    HWSKU=${HWSKU:-`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]'`}
+    PLATFORM=`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`
+    HWSKU=${HWSKU:-`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]' 2>&1`}
+    if [[ $? -ne 0 || HWSKU == "" ]]; then
+            if [[ -f "/etc/sonic/minigraph.xml" ]]; then
+                HWSKU=`sonic-cfggen -m /etc/sonic/minigraph.xml -v "DEVICE_METADATA['localhost']['hwsku']" 2>&1`
+                if [[ $? -ne 0 ]]; then
+                    echo "Failed to get HWSKU"
+                    exit
+                fi
+            fi
+    fi
     /usr/share/sonic/device/$PLATFORM/$HWSKU/$TOPOLOGY_SCRIPT start
 }
 stop() {
     TOPOLOGY_SCRIPT="topology.sh"
-    PLATFORM=${PLATFORM:-`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`}
-    HWSKU=${HWSKU:-`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]'`}
-    usr/share/sonic/device/$PLATFORM/$HWSKU/$TOPOLOGY_SCRIPT stop 
+    PLATFORM=`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`
+    HWSKU=${HWSKU:-`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]' 2>&1`}
+    if [[ $? -ne 0 ]]; then
```

Comment:
Confirm if we need to check for HWSKU == "" as the return value $? would be zero ?

Reply:
Thank you, I have added that check here to ensure that we record all failure conditions.
Also added another check to exit, if HWSKU is not found in config_db and minigraph.xml file is also not available.